presto-main/src/main/java/com/facebook/presto/operator/TaskContext.java

```diff
@@ -159,85 +170,39 @@ public TaskState getState()
 
     public DataSize getMemoryReservation()
     {
-        return new DataSize(memoryReservation.get(), BYTE);
+        return new DataSize(taskMemoryContext.getUserMemory(), BYTE);
```

Comment:
You declared `GuardedBy` for `taskMemoryContext`. But it was never used inside guards.